### PR TITLE
Check if config is nothing before accessing fields

### LINKED ISSUE #280: `ADTypes.AutoMooncake(; config=nothing)` doesn't work
Initialising `ADTypes.AutoMooncake(; config=nothing)` doesn't seem to use a config of `Mooncake.Config()` by default.

The following MWE uses a feature branch of Turing.jl (https://github.com/TuringLang/Turing.jl/pull/2341):

```julia
using Turing
import Mooncake
backend = Turing.AutoMooncake(; config=nothing)  # reexported from ADTypes
@model f() = x ~ Normal(0, 1)
sample(f(), HMC(0.1, 3; adtype=backend), 10)

# works with backend = Turing.AutoMooncake(; config=Mooncake.Config())
```

The error:

```
ERROR: type Nothing has no field debug_mode
Stacktrace:
  [1] getproperty(x::Nothing, f::Symbol)
    @ Base ./Base.jl:37
  [2] ADgradient(x::AutoMooncake{…}, ℓ::LogDensityFunction{…})
    @ MooncakeLogDensityProblemsADExt ~/.julia/packages/Mooncake/eEGqO/ext/MooncakeLogDensityProblemsADExt.jl:63
  [3] ADgradient(ℓ::LogDensityFunction{DynamicPPL.TypedVarInfo{…}, DynamicPPL.Model{…}, DynamicPPL.SamplingContext{…}})
    @ Turing.Inference ~/ppl/lib/src/mcmc/Inference.jl:154
  [4] initialstep(rng::Random.TaskLocalRNG, model::DynamicPPL.Model{…}, spl::DynamicPPL.Sampler{…}, vi_original::DynamicPPL.TypedVarInfo{…}; initial_params::Nothing, nadapts::Int64, kwargs::@Kwargs{})
    @ Turing.Inference ~/ppl/lib/src/mcmc/hmc.jl:159
[...]
```

which points to at least this:

https://github.com/compintell/Mooncake.jl/blob/942eeade0e1d5a9b092f8a9928f8c4807a699487/ext/MooncakeLogDensityProblemsADExt.jl#L61-L72

### PR DESCRIPTION
**Pull request by penelopeysm**
Should fix #280 (I've tested locally), but happy for you to implement a different fix if you think it's better done otherwise. :)

### CODE REVIEW

File: ext/MooncakeLogDensityProblemsADExt.jl [hunk 1]
```diff
@@ -59,8 +59,12 @@ function logdensity_and_gradient(∇l::MooncakeGradientLogDensity, x::Vector{Flo
 end
 
 # Interop with ADTypes.
+function getconfig(x::ADTypes.AutoMooncake)
+    c = x.config
+    isnothing(c) ? Mooncake.DEFAULT_CONFIG : c
```
Comment (willtebbutt) [inline ext/MooncakeLogDensityProblemsADExt.jl]:
```suggestion
    return isnothing(c) ? Mooncake.DEFAULT_CONFIG : c
```
formatting

**Comment by penelopeysm** [comment]
> what's the rationale for introducing a global

To avoid constructing the same value multiple times. I don't know if it makes for any real speedup, but I feel like it makes sense :)

**Comment by penelopeysm** [comment]
Not opposed to that at all. I'll switch those back.

**Comment by willtebbutt** [comment]
Thanks for this. One more thing: could you add a regression test to prevent this error from creeping back in? `test/integration_testing/logdensityproblemsad_interop.jl` would be the place to put it. It could be as simple as more-or-less copying https://github.com/compintell/Mooncake.jl/blob/942eeade0e1d5a9b092f8a9928f8c4807a699487/test/integration_testing/logdensityproblemsad_interop.jl#L23-L25 and replacing the config with `nothing`, or something like that.

Other than that, I think we're good to go.

**Review by willtebbutt — APPROVED** [review]
Thanks for this. I'll merge + tag a release once CI has passed.

**Pull request merged**